### PR TITLE
[redfish] Resolve chassis related ODR

### PR DESCRIPTION
**Pull request by zyoung51**
ODR-795: no blade nodes are contained in the chassis list
- Add blade nodes to chassis list
- Add Contains and ContainedBy member lists

ODR-796: duplicate entries for each enclosure using nodeid and
    chassis serialno
- Remove listing by serial number

ODR-793: ChassisType should reflect Enclosure, Rackmount, or
    Blade as opposed to 'Other'

### CODE REVIEW

File: lib/api/redfish-1.0/chassis.js [hunk 1]
```diff
@@ -10,49 +10,36 @@ var Promise = injector.get('Promise'); // jshint ignore:line
 var Errors = injector.get('Errors');
 var taskProtocol = injector.get('Protocol.Task');
 var controller = injector.get('Http.Services.Swagger').controller;
+var env = injector.get('Services.Environment');
+var nodeApi = injector.get('Http.Services.Api.Nodes');
+
+var EntityIdFilterDefaults = [ '3', '7', '8', '66' ];
```
Comment (yyscamper) [inline lib/api/redfish-1.0/chassis.js]: What's the meaning of these numbers? Please add some comment here.

File: lib/api/redfish-1.0/chassis.js [hunk 2]
```diff
@@ -10,49 +10,36 @@ var Promise = injector.get('Promise'); // jshint ignore:line
...
-            }).then(function(node) {
-                if(!node) {
-                    throw new Errors.NotFoundError('Cound not find node with id ' + identifier);
-                }
+            return nodeApi.getNodeById(identifier)
+            .then(function(node) {
                 return _.filter(node.relations, function(relation) {
                     return relation.relationType === 'encloses';
                 }).map(function(relation) {
                     return relation.targets[0];
```
Comment (yyscamper) [inline lib/api/redfish-1.0/chassis.js]: An enclosure may contains multiple nodes, so relation.targets is an array, and each target is on equal footing and can be in any random order. So what't the reason here only pick the first target?

Comment (zyoung51) [inline lib/api/redfish-1.0/chassis.js]: This is returning a reference to any compute node in the enclosure.  Each target is on equal footing, so the first is picked.

**Comment by yyscamper** [comment]
+1

**Comment by amymullins** [comment]
test this please

**Comment by brianparry** [comment]
test this please

**Comment by amymullins** [comment]
test this please

**Comment by amymullins** [comment]
test this please

**Pull request merged**